langchain/document_loaders/processing/schema.py

```diff
@@ -0,0 +1,92 @@
+"""Schema definitions for document processors.
+
+Document processors take a langchain Document and produce a new Document using some custom
+processing logic.
+"""
+from abc import ABC, abstractmethod
+from typing import List, Sequence
+
+from langchain.schema import Document
+
+
+class BaseDocumentProcessor(ABC):
```

Comment:
we now have DocumentTransformer abstraction that seems very similar, should we try combining?

Reply:
probably yes -- will need to take a look at those

these abstractions will go in last in the sequencing order